### ISSUE TITLE
Loading/disconnected

### ISSUE THREAD

**Issue by nollinvoyd** [RossWang/Aria2-Integration#72]
I have both Aria2 Download Manager Integration 0.4.5 and aria2 1.35.0.  I know aria2 is not for beginners, so I appreciate your patience.  I have several questions.

Does Aria2 Download Manager Integration depend on aria2 1.35.0 to operate? I have it set up, and tried to test it on a YouTube video, but it got stuck here

![aria](https://user-images.githubusercontent.com/20844322/74684083-ca014a00-517f-11ea-92a0-431091e7267d.png)

Perhaps I didn't get the options settings correct?  Please advise.

Thanks

**Comment by no1xsyzy** [comment]
1. You should have a running aria 2 instance, not installed program.
![图片](https://user-images.githubusercontent.com/7346170/74736845-9b25c980-528e-11ea-813a-9ad5202e1acd.png)

2. In firefox addon settings, make this addon link to that aria 2 instance, using IP address/hostname, port, and the token if you have set one. Remember click "Save"
![图片](https://user-images.githubusercontent.com/7346170/74736778-72053900-528e-11ea-8399-59d1e7d2e669.png)
![图片](https://user-images.githubusercontent.com/7346170/74736801-7fbabe80-528e-11ea-824c-98e9df8fa8d7.png)

3. Then open AriaNg it should be working.

**Comment by nollinvoyd** [comment]
This is what i have so far

aria2-1.35.0-win-64bit-build1

![aria cmd](https://user-images.githubusercontent.com/20844322/75120770-f49d4800-5642-11ea-99c9-c738e58ad816.png)

![aria settings](https://user-images.githubusercontent.com/20844322/75120773-ff57dd00-5642-11ea-930a-78af5117b1b4.png)

You said "Then open AriaNg it should be working."  I'm assuming you mean like this?

![toolbar](https://user-images.githubusercontent.com/20844322/75120869-bc4a3980-5643-11ea-8307-df07c53ffa4a.png)

But I still get this

![addon](https://user-images.githubusercontent.com/20844322/75120922-590cd700-5644-11ea-85ad-a49860bad1ce.png)

I also tried this, but all I get is a file called watch

![watch](https://user-images.githubusercontent.com/20844322/75121107-d5ec8080-5645-11ea-92ac-be75c3dee505.png)

**Comment by no1xsyzy** [comment]
no, you need an RPC server instead of a one-time downloader. See `aria2c --help=#rpc`

**Comment by nollinvoyd** [comment]
I don't know how to implement this.  I can't get the syntax right, or the steps.
![aria help](https://user-images.githubusercontent.com/20844322/75379821-dde93200-588a-11ea-9909-1a363dbda978.png)

I ran across this, at https://github.com/ziahamza/webui-aria2

![aria warn](https://user-images.githubusercontent.com/20844322/75380166-6536a580-588b-11ea-9339-93dcc6088f3b.png)

but it says insecure.  

So I have to create a token?  I'm not sure how to go about it.

**Comment by no1xsyzy** [comment]
You need also `--rpc-allow-origin-all` option, if you want to see details from "Details" button. Everything else seems okay and you are ready for adding tasks.
You can enable token authentication with `--rpc-secret=<TOKEN>`.

---
More on token:
**TL;DR**: A token is not forced but highly recommended, just like you would probably lock your house if everyone from all over the world goes everywhere everyday in hundreds of milliseconds.
**Detailed**: Since the server can be easily accessed with HTTP GET, any web page you visit thus its provider (might be evil) is allowed to add/remove a task to/from the aria2 server, even to flood-attack another server by adding many many tasks to your aria2 server. However, with token enabled, the evils must know the token before doing evil things.

**Comment by nollinvoyd** [comment]
Since I already ran it without a tokwn, albeit with a "watch" download result, have I opened/risked a security breach?

I'm still getting the "watch" download

![ariatoken2](https://user-images.githubusercontent.com/20844322/75496155-3a288080-5975-11ea-8856-fdfd20aa5282.png)

![aria config](https://user-images.githubusercontent.com/20844322/75496201-588e7c00-5975-11ea-889b-a1f936f821a6.png)

![aria disconnected](https://user-images.githubusercontent.com/20844322/75496232-74921d80-5975-11ea-9f37-50f198e5a9b5.png)

**Comment by nollinvoyd** [comment]
and this

![webgui](https://user-images.githubusercontent.com/20844322/75498047-04f45080-59be-11ea-9a39-aff839bbd559.png)

![webgui2](https://user-images.githubusercontent.com/20844322/75498151-497fec00-59be-11ea-8342-a91fab4225c0.png)

**Comment by no1xsyzy** [comment]
You need `aria2c --enable-rpc --rpc-listen-all --rpc-allow-origin-all --rpc-secret=<TOKEN>` in one row.
![图片](https://user-images.githubusercontent.com/7346170/75583044-d2268700-5aa7-11ea-891a-2fee04127d41.png)
Then this terminal cannot run any command as intended. Don't close this window or press Ctrl-C to stop it. You will be able to connect with Web-UI, AriaNg or others.

Then you can use any of Web-UIs or Aria2-Integration to add task to it.
![图片](https://user-images.githubusercontent.com/7346170/75583899-85dc4680-5aa9-11ea-999c-002c0ebd26b3.png)

What's more, YouTube videos cannot be simply downloaded this way.
You will only download the web page alone.

**Comment by nollinvoyd** [comment]
Please clarify.  You say YouTube videos cannot be simply downloaded "this way".  Are you saying that aria can only be used for downloading web pages, or that it is just not possible using the webgui?  If possible, please let me know how to go about video download. 

EDIT:  II days ago i posted at [https://github.com/aria2/aria2/issues/1560](url).  I don't know if this is loosely what you may be referring to, but according to the response there "Aria2 is a download manager, it downloads whatever is pointed to by your URL; in this case, a YouTube page (HTML document).  If you want to download a specific resource you see on webpage, (said resource is usually referenced by the HTML document, or a script in the document,) be it a video or an image, you’ll need the URL for that."

When I try to do it thru CMD, I use the "Copy video url at current time" from the video context menu, but, as I said, I only get the "watch" file.

"Then this terminal cannot run any command as intended. Don't close this window or press Ctrl-C to stop it. You will be able to connect with Web-UI, AriaNg or others."

Tho this seemed to succeed, and I left it open, 

![aria final](https://user-images.githubusercontent.com/20844322/75587124-a44f2d00-5a2a-11ea-81b3-b97c94d13627.png)

everything else still fails

![aria after final2](https://user-images.githubusercontent.com/20844322/75587287-fbed9880-5a2a-11ea-914d-7288812042a3.png)

![ariang final](https://user-images.githubusercontent.com/20844322/75587355-20497500-5a2b-11ea-93c0-5171f5c5bd0b.png)

What about settings adjustments

![aria webgui settings2](https://user-images.githubusercontent.com/20844322/75587627-bb424f00-5a2b-11ea-8442-0b3900ce2624.png)

![aria webgui settings1](https://user-images.githubusercontent.com/20844322/75587646-c5fce400-5a2b-11ea-8750-34d18ad1c08a.png)

![aria webgui settings3](https://user-images.githubusercontent.com/20844322/75587666-d90fb400-5a2b-11ea-803f-99311eab38ca.png)

**Comment by no1xsyzy** [comment]
Generally you only need to change the host name and the token.
For host name, if you are running aria 2 instance and browser on same machine, you can use `127.0.0.1`. Otherwise use the IP address of that computer.
For token, you know what you chose.

YouTube is an **online player** not a **video host**. Yes it hosts videos, but what you opens in the browser is a video player written in JavaScript. This video player downloads video from googlevideo.com and plays it. 
"Copy video url at current time" only adds a tag to video player URL telling the player where to start playing.
However, YouTube don't want videos to be easily downloaded, since this will cut their ad fees. You can hardly find anything from google :( There are two project on GitHub to download YouTube videos, [you-get](https://github.com/soimort/you-get) and [youtube-dl](https://github.com/ytdl-org/youtube-dl).
